reporting-app/app/controllers/tasks_controller.rb

```diff
@@ -22,6 +24,22 @@ def assign
     redirect_to task_path(@task)
   end
 
+  # Override parent pick_up_next_task to scope tasks by region.
+  # The parent Strata::TasksController.pick_up_next_task uses Strata::Task.assign_next_task_to,
+  # which doesn't filter by region, so we override to inject policy_scope.
+  def pick_up_next_task
+    # Scope tasks to user's region before finding next unassigned task
+    task = policy_scope(Strata::Task).incomplete.unassigned.first
+
+    if task && task.assign(current_user.id)
+      flash["task-message"] = I18n.t("strata.tasks.messages.task_picked_up")
+      redirect_to task_path(task)
+    else
+      flash["task-message"] = I18n.t("strata.tasks.messages.no_tasks_available")
+      redirect_to tasks_path
+    end
+  end
+
   def request_information
     set_task
```

Comment:
per our convention should be in before action . we set instance vars in callbacks